Flaky test: RingBufferTest.findAndRemoveWorksAcrossThreads times out

## Summary

`RingBufferTest$ConcurrencyTests.findAndRemoveWorksAcrossThreads` fails intermittently with an Awaitility timeout, indicating threads did not finish within the 30-second deadline.

## Error

```
io.camunda.zeebe.logstreams.impl.flowcontrol.RingBufferTest$ConcurrencyTests.findAndRemoveWorksAcrossThreads -- Time elapsed: 30.10 s <<< ERROR!
org.awaitility.core.ConditionTimeoutException:
Condition with alias 'threads complete' didn't complete within 30 seconds because assertion condition
defined as a Lambda expression in io.camunda.zeebe.logstreams.impl.flowcontrol.RingBufferTest$ConcurrencyTests
Expecting value to be false but was true.
    at io.camunda.zeebe.logstreams.impl.flowcontrol.RingBufferTest$ConcurrencyTests.runAndAwait(RingBufferTest.java:551)
    at io.camunda.zeebe.logstreams.impl.flowcontrol.RingBufferTest$ConcurrencyTests.findAndRemoveWorksAcrossThreads(RingBufferTest.java:419)
Caused by: org.opentest4j.AssertionFailedError:
Expecting value to be false but was true
    at io.camunda.zeebe.logstreams.impl.flowcontrol.RingBufferTest$ConcurrencyTests.lambda$runAndAwait$9(RingBufferTest.java:554)
```

## Test Setup

The test spawns a writer thread (puts 100,000 entries into `RingBuffer`) and a processor thread (calls `findAndRemove` for each position in order, spinning with `LockSupport.parkNanos(1)` until found). `runAndAwait` uses a 30-second Awaitility timeout asserting both threads have terminated.

The timeout indicates at least one thread was still alive and not making progress after 30 seconds, rather than being slow — the processor spin loop has no exit path other than finding the entry.

## Link
- [Failed Github Action](https://github.com/camunda/camunda/actions/runs/25921590410/attempts/1)

--- a/zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RingBufferTest.java
+++ b/zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RingBufferTest.java
@@ -378,7 +378,9 @@ final class RingBufferTest {
 
     /**
      * Simulates the full lifecycle: writer puts entries, a second thread finds and removes them
-     * (like onProcessed). Verifies findAndRemove works correctly.
+     * (like onProcessed). The processor only advances once the writer has published the position,
+     * mirroring the production ordering where onProcessed cannot outrun the producer. Verifies
+     * findAndRemove works correctly.
      */
     @Test
     void findAndRemoveWorksAcrossThreads() {
@@ -386,6 +388,7 @@ final class RingBufferTest {
       final var buffer = new RingBuffer(CAPACITY);
       final var failures = new ConcurrentLinkedQueue<Throwable>();
       final var startLatch = new CountDownLatch(1);
+      final var publishedPosition = new AtomicLong(0);
 
       final var writer =
           new Thread(
@@ -393,6 +396,7 @@ final class RingBufferTest {
                 awaitLatch(startLatch);
                 for (long pos = 1; pos <= ITERATIONS; pos++) {
                   buffer.put(newEntry(pos));
+                  publishedPosition.set(pos);
                 }
               });
 
@@ -401,6 +405,10 @@ final class RingBufferTest {
               () -> {
                 awaitLatch(startLatch);
                 for (long pos = 1; pos <= ITERATIONS; pos++) {
+                  while (pos > publishedPosition.get()) {
+                    LockSupport.parkNanos(1);
+                  }
+
                   InFlightEntry entry = null;
                   while (entry == null) {
                     entry = buffer.findAndRemove(pos);
@@ -547,7 +555,7 @@ final class RingBufferTest {
       startLatch.countDown();
 
       Awaitility.await("threads complete")
-          .atMost(Duration.ofSeconds(30))
+          .atMost(Duration.ofSeconds(60))
           .untilAsserted(
               () -> {
                 for (final var thread : threads) {

